colossalai/inference/kv_cache/kvcache_manager.py

```diff
@@ -221,11 +221,14 @@ def allocate_single_block(self, block_table: torch.Tensor, block_local_idx: int,
         Returns:
             The remaining space required to be allocated (in other blocks).
         """
-        assert block_table.dim() == 1
+        space_asked = 1
         block_global_id = block_table[block_local_idx].item()
         if block_global_id < 0:
             # Allocate a new block if the current position is not assigned a block yet
-            assert self._available_blocks > 0, "No available blocks to allocate."
+            if self._available_blocks <= 0:
+                # No available blocks to allocate, we free current sequence and return it to
+                self.free_block_table(block_table)
+                return True
```

Comment:
Shouldn't it be False here, and use alloc_failed instead of recycle for the variable name above? This is easy to make reader confuse.

Reply:
but the following return `0` when running successfully, and `possitive int value` when failed